kart/completion.py

```diff
@@ -15,6 +15,10 @@
 except ImportError:
     shellingham = None
 
+def provide_default_shell():
+    if os.name == 'posix':
+        return os.environ['SHELL']
```

Comment:
from https://github.com/sarugaku/shellingham but safeguarded against KeyErrors. The caller already maps '' to a clean user error. 

```suggestion
    if os.name == 'posix':
        return os.environ.get('SHELL', '')
    elif os.name == 'nt':
        return os.environ.get('COMSPEC', '')
```

Reply:
Sounds good, I had left of the nt version since we don't currently support helper mode there and so shouldn't have to fall back to a default way of finding the shell.

Reply:
Not sure how we'd get to test that.

Reply:
> I had left of the nt version since we don't currently support helper mode there and so shouldn't have to fall back to a default way of finding the shell.

Ah, ok. TBC, tab completion _does_ work on Windows

Reply:
As below I've added the test in the powershell e2e test and that picked up the need for an additional hidden import though it sounds like the env var will always be present on windows so that COMSPEC check might be safe.

Reply:
so, we should add the fallback to COMSPEC? is gone in the latest patch

Reply:
Yep. Will do this evening.